src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEMethodSymbol.cs

```diff
@@ -1221,6 +1215,38 @@ public override ImmutableArray<MethodSymbol> ExplicitInterfaceImplementations
             }
         }
 
+        private ImmutableArray<MethodSymbol> ExplicitlyOverriddenMethods()
+        {
+            var moduleSymbol = _containingType.ContainingPEModule;
+
+            // Context: we need the containing type of this method as context so that we can substitute appropriately into
+            // any generic interfaces that we might be explicitly implementing.  There is no reason to pass in the method
+            // context, however, because any method type parameters will belong to the implemented (i.e. interface) method,
+            // which we do not yet know.
+            return new MetadataDecoder(moduleSymbol, _containingType).GetExplicitlyOverriddenMethods(_containingType.Handle, _handle, this.ContainingType);
+        }
+
+        internal MethodSymbol ExplicitlyOverriddenClassMethod
+        {
+            get
+            {
+                if (IsExplicitClassOverride)
+                {
+                    foreach (MethodSymbol method in ExplicitlyOverriddenMethods())
```

Comment:
>ExplicitlyOverriddenMethods [](start = 52, length = 27)

I think it would make sense to bundle this work with what ```ExplicitInterfaceImplementations``` property is doing to avoid decoding metadata over and over again. And I think it would make sense to cache/reuse the result.

Reply:
I thought of that. But this will only be called when `IsExplicitClassOverride` is set, which is rarely.  I'd rather not increase the size of a `PEMethodSymbol` until there is evidence that such a cache would be needed often enough to not be a drag on performance.

---
In reply to: [421732856](https://github.com/dotnet/roslyn/pull/43797#discussion_r421732856) [](ancestors = 421732856)

Reply:
> I thought of that. But this will only be called when IsExplicitClassOverride is set, which is rarely.  

I don't think this is accurate. This is property, ```ExplicitlyOverriddenClassMethod```, can be called any time. And every time that happens and ```IsExplicitClassOverride``` is set (BTW, we are going to decode the same metadata in order to figure out if it is set), we are going to decode metadata. For every access to this property.

> I'd rather not increase the size of a PEMethodSymbol

We already have a solution for that - ```_uncommonFields``` are used to cache uncommon things.

---
In reply to: [421770271](https://github.com/dotnet/roslyn/pull/43797#discussion_r421770271) [](ancestors = 421770271,421732856)

Reply:
`IsExplicitClassOverride` is already cached, and is rarely set.

---
In reply to: [421859886](https://github.com/dotnet/roslyn/pull/43797#discussion_r421859886) [](ancestors = 421859886,421770271,421732856)